include/nbl/system/ISystem.h

```diff
@@ -61,10 +64,11 @@ class NBL_API2 ISystem : public core::IReferenceCounted
         {
             E_REQUEST_TYPE type;
             std::variant<
+                SRequestParams_EMPTY,
                 SRequestParams_CREATE_FILE,
                 SRequestParams_READ,
                 SRequestParams_WRITE
-            > params;
+            > params = SRequestParams_EMPTY();
```

Comment:
this doesn't have a default constructor so its breaks a couple of classes that have a default construtor. I guess this is related to gnu implementation of STL 


I guess SRequestParams_CREATE_FILE does not have a default constructor? 

https://en.cppreference.com/w/cpp/utility/variant/variant

```
1) Default constructor. Constructs a variant holding the [value-initialized](https://en.cppreference.com/w/cpp/language/value_initialization) value of the first alternative ([index()](https://en.cppreference.com/w/cpp/utility/variant/index) is zero).

    This constructor is constexpr if and only if the value initialization of the alternative type T_0 would satisfy the requirements for a [constexpr function](https://en.cppreference.com/w/cpp/language/constexpr).
    This overload participates in overload resolution only if [std::is_default_constructible_v](http://en.cppreference.com/w/cpp/types/is_default_constructible)<T_0> is true.
```